app/v7/triggerCalculation/TriggerCalculationService.scala

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v7.triggerCalculation
+
+import shared.controllers.RequestContext
+import shared.models.errors._
+import api.errors._
+import shared.services.{BaseService, ServiceOutcome}
+import cats.implicits._
+import v7.triggerCalculation.model.request.TriggerCalculationRequestData
+
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class TriggerCalculationService @Inject() (connector: TriggerCalculationConnector) extends BaseService {
+
+  def triggerCalculation(request: TriggerCalculationRequestData)(implicit
+      ctx: RequestContext,
+      ec: ExecutionContext): Future[ServiceOutcome[Unit]] = {
+
+    connector.triggerCalculation(request).map(_.leftMap(mapDownstreamErrors(downstreamErrorMap)))
+  }
+
+  private val downstreamErrorMap: Map[String, MtdError] = {
+    val errors = Map(
+      "INVALID_NINO"            -> NinoFormatError,
+      "INVALID_TAX_YEAR"        -> TaxYearFormatError,
+      "INVALID_TAX_CRYSTALLISE" -> FinalDeclarationFormatError,
+      "INVALID_REQUEST"         -> InternalError,
+      "NO_SUBMISSION_EXIST"     -> RuleNoIncomeSubmissionsExistError,
+      "CONFLICT"                -> RuleFinalDeclarationReceivedError,
```

Comment:
This is where you need to add the new errors:
```suggestion
      "CONFLICT"                -> RuleFinalDeclarationReceivedError,
      "OUTSIDE_AMENDMENT_WINDOW"       -> RuleOutsideAmendmentWindowError,
      "DECLARATION_NOT_RECEIVED" -> RuleDeclarationNotReceivedError,
```